features/TESTS/filesystem/heap_block_device/main.cpp

```diff
@@ -0,0 +1,97 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+
+#include "HeapBlockDevice.h"
+#include <stdlib.h>
+
+using namespace utest::v1;
+
+/* It is not possible to build a KL25Z image with IAR including the file system if
+ * stack tracking statistics are enabled. If this is the case, build dummy
+ * tests.
+ */
+#if ! defined(TOOLCHAIN_IAR) && ! defined(TARGET_KL25Z) && ! defined(MBED_STACK_STATS_ENABLED)
```

Comment:
In the past we have split up tests due to their size, so I'd say this is an acceptable solution if there is a reasonable way to split this up.

A test *definitely* should not change its behavior due to the target or the toolchain. So the way it is currently implemented should change.

Could we not simply reduce the size of the `HeapBlockDevice`? Does it have to be `16*BLOCK_SIZE`? That's fairly large (8k if I'm doing my math right). The KL25Z only has 16K of RAM.